src/OpenTelemetry.Extensions.Hosting/README.md

```diff
@@ -83,6 +83,28 @@ app.Run();
 A fully functional example can be found
 [here](https://github.com/open-telemetry/opentelemetry-dotnet/tree/main/examples/AspNetCore).
 
+### Resources
+To dynamically add resources at startup from the dependency injection you can provide an `IResourceDetector`.
+To make use of it add it to the dependency injection and they you can use the `ISerivceProvider` add it to OpenTelemetry:
+```csharp
+public class YourResourceDetector : IResourceDetector
```

Comment:
Do we want to use "your" or "my"? (either way works for me as long as we are consistent) https://github.com/open-telemetry/opentelemetry-dotnet/blob/0971088451eb0ec8cef06b852d86cff640e4ec6d/docs/trace/extending-the-sdk/MyExporter.cs#L21

Reply:
I don't care honestly, I always feel more like the documentation is speaking to me, not that the example code is mine, but let's keep it consistent.